src/uint256.h

```diff
@@ -88,12 +89,30 @@ class base_blob
     }
 };
 
+namespace detail {
+/**
+ * Writes the hex string into a new uintN_t object and only returns a value
+ * iff all of the checks pass:
+ *   - All characters are hex
+ *   - Input length is uintN_t::size()*2
+ */
+template <class uintN_t>
+std::optional<uintN_t> FromHex(std::string_view str)
+{
+    if (uintN_t::size() * 2 != str.size() || !IsHex(str)) return std::nullopt;
+    uintN_t rv;
+    rv.SetHexFragile(str);
```

Comment:
after we get rid of the deprecation, these checks should ideally be done while iterating over the values to avoid multiple iterations (e.g. how Base58 parsing is done)

Reply:
Yes, definitely different PR